### ISSUE TITLE
false positive Incompatible types in assignment

### ISSUE THREAD

**Issue by sam-s** [python/mypy#21577]
The following idiomatic pandas code
```
     logvol : pd.Series[float] = df.loc["ZZZ",my_col]
     logvol = np.log10(logvol[logvol > 0])
```
triggers
```
z.py:261: error: Incompatible types in assignment (expression has type "str | bytes | date | datetime | timedelta | <7 more items> | complex | integer[Any] | floating[Any] | complexfloating[Any, Any]", variable has type "Series[float]")  [assignment]
z.py:262: error: Incompatible types in assignment (expression has type "ndarray[tuple[Any, ...], dtype[Any]]", variable has type "Series[float]")  [assignment]
```
because, 
1. my type annotation is ignored (`.loc` can be a scalar)
2. `mypy` is unaware that `numpy` returns pandas objects for pandas argument

the alternative, longer and less idiomatic code
```
    logvol = df[my_col].xs("ZZZ")
    logvol = logvol[logvol > 0].map(np.log10)
```
produces no diagnostics.

**Your Environment**

- Mypy version used: mypy 2.1.0 (compiled: yes)
- Mypy command-line flags:  --install-types --non-interactive --check-untyped-defs
- Mypy configuration options from `pyproject.toml` (and other config files):
```
[tool.mypy]
disable_error_code = [
    "import-untyped",
    "valid-type",
    "var-annotated",
]
[[tool.mypy.overrides]]
module = ["PyQt6.*"]
follow_imports = "skip"
```
- Python version used:  3.14.5

**Comment by ilevkivskyi** [comment]
Same story here, please show that this is an actual `mypy` bug, rather than bad/incomplete stubs/annotations in relevant 3rd party libraries.

**Comment by sam-s** [comment]
> Same story here, please show that this is an actual `mypy` bug, rather than bad/incomplete stubs/annotations in relevant 3rd party libraries.

I know way too little about either mypy or "stubs/annotations" to even contemplate looking into this.
You are in a better position to route this problem.
Thank you.
